src/reports/test.js

```diff
@@ -75,4 +75,14 @@ export default class TestReport {
 	getAssertions() {
 		return this.assertions.slice();
 	}
+
+	// Remove actual and expected values from assertions. This is to prevent
+	// leaking memory throughout a test suite.
+	slimAssertions() {
+		this.assertions = this.assertions.map( assertion => {
```

Comment:
I personally prefer re-assignment even if it is doing the same thing as it makes it explicit that mutation occurs during the loop.

As for a test, sounds like a good idea. I'll look into it.